src/app.js

```diff
@@ -9,23 +12,32 @@ class App extends React.Component {
         button: 'get from query',
       },
       query: '["metrosById", [72], ["name"]]',
+      queries: [],
       response: {},
       error: {},
     }
 
     this.handleOnChange = this.handleOnChange.bind(this)
     this.handleOnClick = this.handleOnClick.bind(this)
+    this.updateQuery = this.updateQuery.bind(this)
   }
 
-  componentDidMount() {
+  componentDidMount(){
+    this.setState({queries: Lockr.get('queries', [])})
     this.falcorGet()
   }
 
-  handleOnChange(event) {
-    this.setState({ query: event.target.value })
-  }
+  updateQuery(query){ this.setState({ query }) }
+  handleOnChange(event){this.updateQuery(event.target.value) }
 
-  handleOnClick() {
+  handleOnClick(){
+    const queries = this.state.queries
+    const query = this.state.query
+    if (queries.length === 0 || queries[queries.length - 1] !== query) {
+      const updatedQueries = queries.concat(query)
+      this.setState({queries:updatedQueries })
+      Lockr.set('queries', updatedQueries)
+    }
```

Comment:
basically every time you "press the button" you add the query in the stack right?